library/Zend/Stdlib/Hydrator/Strategy/StrategyChain.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Stdlib\Hydrator\Strategy;
+
+use Traversable;
+use Zend\Stdlib\PriorityQueue;
+
+final class StrategyChain implements StrategyInterface
+{
+    /**
+     * Default priority at which strategies are added
+     */
+    const DEFAULT_PRIORITY = 1;
+
+    /**
+     * Strategy chain
+     *
+     * @var PriorityQueue
+     */
+    private $strategies;
+
+    /**
+     * Initialize Strategy chain
+     *
+     * @param array|Traversable $strategies
+     */
+    public function __construct($strategies)
+    {
+        $this->strategies = new PriorityQueue();
+        $this->setStrategies($strategies);
+    }
+
+    /**
+     * Sets strategies
+     *
+     * @param  array|Traversable                  $strategies
+     * @return void
+     * @throws Exception\InvalidArgumentException
+     */
+    private function setStrategies($strategies)
+    {
+        if (!is_array($strategies) && !$strategies instanceof Traversable) {
```

Comment:
You can probably skip all this and use `ArrayUtils::iteratorToArray()` instead, which already encapsulates this conditional